packages/static-module-record/src/babelPlugin.js

```diff
@@ -536,18 +537,33 @@ function makeModulePlugins(options) {
             const { local, exported } = spec;
             const importFrom =
               spec.type === 'ExportNamespaceSpecifier' ? '*' : local.name;
-
+            let myUpdaterSources;
             // If local.name is reexported we omit it.
             const importTo = exported.name;
-            let myUpdaterSources = updaterSources[importFrom];
-            if (myImportSources) {
-              myUpdaterSources = myImportSources[importFrom];
-              if (!myUpdaterSources) {
-                myUpdaterSources = [];
-                myImportSources[importFrom] = myUpdaterSources;
+
+            if (source) {
+              // If source is defined, it's a reexport from a different module.
+              // In this case we want to collect the local and reexported name
+              // If there's no `as newName`, the ExportNamedDeclaration will have the `exported` field anyway,
+              // representing the same identifier as `local`
+
+              if (!reexportMap[source.value]) {
+                reexportMap[source.value] = [];
+              }
+              reexportMap[source.value].push([local.name, exported.name]);
```

Comment:
Used the source name as key here instead of exported name, because I need to iterate the reexportMap once to get all exports and I'd need to keep iterating it in module-instance if it was keyed on exports.